src/Functions/FunctionsExternalDictionaries.h

```diff
@@ -971,6 +971,14 @@ class FunctionDictGet final : public IFunction
             const auto & key_columns = assert_cast<const ColumnTuple &>(*key_col).getColumnsCopy();
             const auto & key_types = static_cast<const DataTypeTuple &>(*key_col_with_type.type).getElements();
 
+            assert(key_columns.size() == key_types.size());
+            const auto & structure = dict->getStructure();
+            assert(structure.key);
+            size_t key_size = structure.key->size();
+            if (key_columns.size() != key_size)
```

Comment:
It seems it doesn't work for polygon dictionaries because they have `Array(Array(Array(Array(Float64))))` as key of size 1. And third argument in dictGet is tuple(x, y) of size 2.